.ci/scripts/test_synapse_port_db.sh

```diff
@@ -45,9 +45,23 @@ rm .ci/test_db.db
 poetry run update_synapse_database --database-config .ci/sqlite-config.yaml --run-background-updates
 
 # re-create the PostgreSQL database.
-poetry run .ci/scripts/postgres_exec.py \
-  "DROP DATABASE synapse" \
-  "CREATE DATABASE synapse"
+psql \
+  -c "DROP DATABASE synapse" \
+  -c "CREATE DATABASE synapse"
 
 echo "+++ Run synapse_port_db against empty database"
 poetry run synapse_port_db --sqlite-database .ci/test_db.db --postgres-config .ci/postgres-config.yaml
+
+echo "--- Create a brand new postgres database from schema"
+cp .ci/postgres-config.yaml .ci/postgres-config-unported.yaml
+sed -i -e 's/database: synapse/database: synapse_unported/' .ci/postgres-config-unported.yaml
+psql -c "CREATE DATABASE synapse_unported"
+poetry run update_synapse_database --database-config .ci/postgres-config-unported.yaml --run-background-updates
+
+echo "+++ Comparing ported schema with unported schema"
+# Ignore the tables that portdb creates. (Should it tidy them up when the porting is completed?)
```

Comment:
hmm. What do you call 'completed' though?; the script is meant to be re-run multiple times.
I think it's OK as it is. It may be useful to have a trace that the database is a ported one anyway.

Reply:
> What do you call 'completed' though?; the script is meant to be re-run multiple times.

My understanding: portdb is a one-time migration process that can be suspended (cancel the script) and resumed (rerun the script). Once it finished porting across all rows you're supposed to only write to the postgres DB, and not rewrite to the sqlite db.

More precisely: the portdb script assumes the following for each table to be ported.

> If all rows in the range X <= rowid <= Y have been ported in a previous run 1 of the script, then there are no new rows in that range added between run 1 and run 2 of the script.

I am slightly paranoid that this assumption is false. https://sqlite.org/rowidtable.html notes that rowids can change if you vacuum the script. If you have an `INTEGER PRIMARY KEY` then you can insert any rowid you like in any order you like. (Related: https://github.com/matrix-org/synapse/pull/13226.)

Reply:
To be explicit: the above is mostly paranoia and extreme caution rather than anything concrete. I plan to leave the script and comment as it is. How does that sound?

Reply:
My understanding was that you can re-run the portdb script many times online (it's an incremental process), but then you finalise it by running it once again with Synapse offline. Hence there isn't really a 'completion' condition, unless we make the admin specify some flag to say 'this is the last time I intend to run the script'..